### ISSUE TITLE
What is unit quaternion?

### ISSUE THREAD

**Issue by eastskykang** [CMM-22/a5#2]
Please find a short description about unit quaternion here. 

**TL;DR:**

In short, unit quaternion is one way of representing 3D rotational transformation (or 3D orientation of a coordinate system). Its real part is ```cos (0.5 * theta)``` and its imaginary part is ```sin(0.5 * theta) * v ``` where (theta, v) is an angle of rotation and an axis of rotation (in an unit vector) respectively; for the axis vector [x, y, z], we define ```v := xi + yj + zk```. 

For instance, let's say you rotate some vector around x axis by 1 rad. The corresponding quaternion is ```q = [cos(0.5), sin(0.5), 0, 0, 0]``` or more specifically, ```cos(0.5) + sin(0.5) i + 0 j + 0 k```

Then you can use this quaternion ```q``` to rotate a 3D vector v around x axis by 1 rad with some special mathematical operation (option 1 is using matrix operation, option 2 is using complex number multiplication. see the one of the references below.)

If you want some materials to learn more about quaternion. I recommend these amazing videos from 3brown1blue: 
- https://www.youtube.com/watch?v=d4EgbgTm0Bg
- https://www.youtube.com/watch?v=zjMuIxRvygQ
- it's not about quaternion but the linear algebra lectures of 3Brown1Blue could be very useful for this course in general, so I do recommend you to watch it

Also this webpage:
- https://eater.net/quaternions

Other references are here
- https://en.wikipedia.org/wiki/Quaternion
- https://en.wikipedia.org/wiki/Quaternions_and_spatial_rotation 
- https://ethz.ch/content/dam/ethz/special-interest/mavt/robotics-n-intelligent-systems/rsl-dam/documents/RobotDynamics2018/RD_HS2018script.pdf (see page 19 - )

------
For now, in our code base, you can just use quaternion as the same way you use a rotation matrix.

For instance, let's say Eigen quaternion ```qAB``` is equivalent to a rotational matrix R_{A}{B} where```v_A = R_{A}{B} v_B```; v_A denotes a 3D vector **v** expressed in frame A, and v_B denotes **v** expressed in frame B .

Then, in cpp code, you can use quaternion as follows

```cpp
V3D v_A = qAB * v_B
```

Please note that this is not just a simple matrix-vector or vector-vector multiplication but rather a special operation for rotating a vector with an unit quaternion that is already implemented and overloaded by cpp * operator in Eigen library. 
I hope this helps.
